AEPCore/Sources/eventhub/SDKInstanceServiceProvider.swift

```diff
@@ -0,0 +1,136 @@
+/*
+ Copyright 2024 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+*/
+
+import AEPServices
+
+/// provides all the services needed by an `Extension` which supports multi-instances.
+@objc(AEPSDKInstanceServiceProvider)
+public class SDKInstanceServiceProvider: NSObject {
+
+    private let identifier: SDKInstanceIdentifier
+    private let logger: Logger
+
+    private var dataStoreInstances: [String: NamedCollectionDataStore] = [:]
```

Comment:
IMO, there’s no need to cache any of these. DataQueue is already cached by ServiceProvider, and Cache and NamedCollection are just wrappers around the actual services.